### PR TITLE
Adding data ingest for global ensembles. 

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 
Adds functionality to `ush/retrieve_data.py` that supports ingesting global ensemble members in the ways that have been used for RRFS. It also adds GEFS and GDAS known data locations to `ush/templates/data_locations.yml`. This means that it will pull data from AWS and/or HPSS for:
- GEFS ICS and LBCS grib2 a & b files (it does not merge them, as that is outside the scope of this tool's functionality). 
- GDAS ICS and LBCS netcdf files. For ICS, it needs atm and sfc files. For LBCS, it only needs atm files.

The tool is set up to pull all members as a single job, or it can be configured to do a subset of members if the user prefers to pull subsets of members. Just like for other datasets, the ICS and LBCS should be handled in separate calls to the tool, and the tool only handles a single start time since that is consistent with the behavior of the workflow.

Additional functionality now includes the ability to template the output directory to include the ensemble member so each member can be stored in its own separate directory when the workflow runs.

I have added functional tests to pull data from AWS and HPSS, and have hooked the AWS tests up to GitHub Actions to run when the relevant files have changed. No user intervention is needed to have those tests run -- they should run automatically upon opening a PR.

## TESTS CONDUCTED: 
I have ensured that all the existing functional tests run on Hera for AWS and HPSS.
I still need to test that the workflow still runs as expected with the modifications to the tool. I will report back on those tests upon their completion.

## DEPENDENCIES:
None.

## CONTRIBUTORS (optional): 
@robgonzalezpita contributed significantly to getting this functionality in place.

### CODE REVIEW

**Review by robgonzalezpita — COMMENTED** [review]
This looks really good. Nice job on refactoring the script and adding the unit tests!

After running the full test suite on hera: 

```
Ran 9 tests in 1488.575s
FAILED (failures=2, errors=1)
```
The two failed tests are `test_rap_lbcs_from_aws` ( only fhr 21  )  &  
`test_hrrr_lbcs_from_aws` ( only fhr 15 ). 

The test which errors is `test_hrrr_ics_from_aws`.  I changed the `--cycle_date` to `2022062500`, ran the single test, and it passed.

File: ush/test_retrieve_data.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+class FunctionalTesting(unittest.TestCase):
+
+    ''' Test class for retrieve data '''
+
+    def setUp(self):
+        self.path = os.path.dirname(__file__)
+        self.config = f'{self.path}/templates/data_locations.yml'
+
+    def test_fv3gfs_lbcs_from_hpss(self):
```
Comment (robgpita) [inline ush/test_retrieve_data.py]: What are your thoughts on adding the line `@unittest.skipIf(os.environ['CI'] == "true", "Skipping HPSS tests")` above the function definitions which pull from hpss, to skip running tests which we know will fail when on the GHA runner?

Comment (christinaholtNOAA) [reply ush/test_retrieve_data.py]: I really like that solution! 

I knew I needed to come back to address the issue of not running the HPSS tests here. That would work!

**Comment by panll** [comment]
I got following errors when testing on Hera:
---------------------------------------------
Traceback (most recent call last):
  File "/scratch2/BMC/det/lpan/review/08012022/test/ufs-srweather-app/regional_workflow/ush/retrieve_data.py", line 963, in <module>
    main(sys.argv[1:])
  File "/scratch2/BMC/det/lpan/review/08012022/test/ufs-srweather-app/regional_workflow/ush/retrieve_data.py", line 806, in main
    unavailable = hpss_requested_files(
  File "/scratch2/BMC/det/lpan/review/08012022/test/ufs-srweather-app/regional_workflow/ush/retrieve_data.py", line 551, in hpss_requested_files
    unavailable[existing_archive] = set(clean_up_output_dir(
TypeError: 'NoneType' object is not iterable
+ print_err_msg_exit 'Call to retrieve_data.py failed with a non-zero exit status.
------------------------------------------------------------------------------------------
Here is the log file: hfe04:/scratch2/BMC/det/lpan/review/08012022/test/expt_dirs/test_community/log/get_extrn_ics_2019061518.log
@christinaholtNOAA

**Comment by christinaholtNOAA** [comment]
@panll Thanks for pointing that out. I believe I have addressed that error with my latest commit. I ran a limited set of WE2E tests that also passed on Hera. I will kick off the auto tests to make sure those run now.

**Comment by venitahagerty** [comment]
Machine: hera
Compiler: intel
Job: WE
Repo location: /scratch1/BMC/zrtrr/rrfs_ci/autoci/pr/983954195/20220812000519/ufs-srweather-app
Build was Successful
Rocoto jobs started
Long term tracking will be done on 9 experiments
If test failed, please make changes and add the following label back:
ci-hera-intel-WE

**Comment by venitahagerty** [comment]
Machine: jet
Compiler: intel
Job: WE
Repo location: /lfs1/BMC/nrtrr/rrfs_ci/autoci/pr/983954195/20220812000515/ufs-srweather-app
Build was Successful
Rocoto jobs started
Long term tracking will be done on 9 experiments
If test failed, please make changes and add the following label back:
ci-jet-intel-WE

**Comment by danielabdi-noaa** [comment]
This looks good to me. Also, I am glad to see tests on Jet passing.

**Review by danielabdi-noaa — APPROVED** [review]

**Comment by JeffBeck-NOAA** [comment]
@linlin-pan, can you confirm if your test is now successful?  Thanks.

**Comment by panll** [comment]
Yes, it works now. Thanks! @JeffBeck-NOAA @christinaholtNOAA

**Review by panll — APPROVED** [review]

**Review by JeffBeck-NOAA — APPROVED** [review]

**Pull request merged**